packages/rum-core/src/browser/cookieObservable.ts

```diff
@@ -0,0 +1,68 @@
+import type { Configuration, CookieStore } from '@datadog/browser-core'
+import {
+  setInterval,
+  clearInterval,
+  Observable,
+  addEventListener,
+  ONE_SECOND,
+  findCommaSeparatedValue,
+  DOM_EVENT,
+} from '@datadog/browser-core'
+
+export interface CookieStoreWindow extends Window {
+  cookieStore?: CookieStore
+}
+
+export type CookieObservable = ReturnType<typeof createCookieObservable>
+
+export type CookieChange = { name: string; value: string | undefined }
+
+export function createCookieObservable(configuration: Configuration, cookieName: string) {
+  const detectCookieChangeStrategy = (window as CookieStoreWindow).cookieStore
+    ? listenToCookieStoreChange(configuration)
+    : watchCookieFallback
+
+  return new Observable<CookieChange>((observable) =>
+    detectCookieChangeStrategy(cookieName, (event) => observable.notify(event))
+  )
```

Comment:
**🥜 nitpick:** ‏For the sake of readability, I would rewrite this like this:

```suggestion
  return new Observable<CookieChange>((observable) => {
    if ((window as CookieStoreWindow).cookieStore) {
      return listenToCookieStoreChange(configuration, cookieName, observable.notify)
    }
    return watchCookieFallback(cookieName, observable.notify)
  })
```

Reasoning: this removes a few indirections ('Strategy' variable, currying/intermediary function)

Reply:
Well I made this change because of this comment: https://github.com/DataDog/browser-sdk/pull/2639#discussion_r1544284211

Reply:
Clarity is subjective, I stand by my opinion, but feel free to dismiss 😃  This is just a nitpick